scheduler/reconcile.go

```diff
@@ -304,12 +304,17 @@ func (a *allocReconciler) markStop(allocs allocSet, clientStatus, statusDescript
 // markDelayed does markStop, but optionally includes a FollowupEvalID so that we can update
 // the stopped alloc with its delayed rescheduling evalID
 func (a *allocReconciler) markDelayed(allocs allocSet, clientStatus, statusDescription string, followupEvals map[string]string) {
+	var e string
```

Comment:
I think this will be easier to reason about if the `e` is inside the loop.  I'd fear that if an alloc doesn't have a follow up eval associated with it (not sure if that's expected, or due to a bug), the result will contain follow up evals of a previous element in list.

Also, an alternative to initialize `followUpEvals` to an empty map if nil, and you avoid the conditionals inside the loop.

Reply:
I think empty initialization is correct, you're right about assigning the wrong `followupEval` to the allocation, only some allocations will have keys in the map. thanks, fixed.